main/streams/plain_wrapper.c

```diff
@@ -785,7 +786,19 @@ static int php_stdiop_set_option(php_stream *stream, int option, int value, void
 							return PHP_STREAM_OPTION_RETURN_ERR;
 						}
 
-						size = GetFileSize(hfile, NULL);
+						if (!GetFileSizeEx(hfile, &file_size)) {
+							return PHP_STREAM_OPTION_RETURN_ERR;
+						}
+# if defined(_WIN64)
+						size = file_size.QuadPart;
+# else
+						if (file_size.HighPart) {
+							/* file is too large; process as much as possible */
+							size = SIZE_MAX;
```

Comment:
Will this still copy the rest as a separate chunk, or produce a partial copy?

Reply:
That would copy only `SIZE_MAX` bytes, but copy() returns false in that case.

Reply:
What would happen on a 32bit POSIX system? Would `stat()` fail for 4GiB files?

Reply:
At least according to POSIX it should fail (likely for >= 2 GiB already), so I matched the implementation; that doesn't make a real differrence in practise, though.